### ISSUE TITLE
GraphQL support

### ISSUE THREAD

**Issue by rafsanulhasan** [simplcommerce/SimplCommerce#871]
Hi @thiennn,
Hope you are doing well.

I tried it with a GraphQL and succeeded...
Even we can try gRPC also 

GraphQL will let us decrease data transfers hugely since it gives us the ability to only return the data we need. Additionally, we can bring real time functionality for some modules like Ordering , Order tracking and so on....

**Comment by thiennn** [comment]
Hi @rafsanulhasan ,

I think it is not worth supporting that. Rest API still serve us well and everyone familiar with rest.

Realtime support is another story

**Comment by rafsanulhasan** [comment]
Hi @thiennn,
IMHO, I think its worth supporting RPC in addition to REST because performance matters a lot in real life scenarios because some devices consume device resources a lot. When client want to host this app into a shared hosting, where resources are limited, they can run into a huge performance issue which I faced when I deployed this app (a bit modified based on client requirement) into a shared server for a client.   

Then I had to apply some mechanisms to decrease the amount of data transfer, Hence, I refactored the APIs to add support for both RPC (graphql) and REST. Now all my clients are happy. However, it needed a small effort to refactor and add graphql support into this project, 

I want to work on this. 

Should I open a separate issue for real-time support?
